operatorcourier/validate.py

```diff
@@ -371,9 +371,8 @@ def is_email(field):
             return True
 
         def is_version(field):
-            pattern1 = re.compile(r'v(\d+\.)(\d+\.)(\d)')
-            pattern2 = re.compile(r'(\d+\.)(\d+\.)(\d)')
-            return pattern1.match(field) or pattern2.match(field)
+            pattern = re.compile(r'(v)?(\d+\.)(\d+\.)(\d)(-[a-z0-9\.])*')
```

Comment:
Do you need all `()`? You are not returning any substring from field

```python
r'v?\d+[.]\d+[.]\d(-([a-z0-9][.]?)+)*'
```
I miss unittests.
Actually this is not working as you expect. `(-[a-z0-9\.])*` matches `-a-.-2` instead `-a.2`. How about uppercase letters and underscores, are they allowed?
Also If I remember correctly quay doesn't allow leading zeroes

Reply:
For the AppRegistry, quay allows any valid semver. Things like `0.0.1` and `1.21.5-beta1` are valid.

Reply:
I agree that we can remove extraneous groups. I'd actually prefer using a proven semver library rather than reinventing the regex.

Reply:
Scratch what I said, this version in the CSV has no relation to quay's AppRegistry, my mistake.
In any case, there should be a doc that outlines what format this version field can take. From experience I have seen versions that use a semver followed by a tag. I agree that `-a-.-2` doesn't look right, so I'll update it to avoid something like that

Reply:
It'd be better to use an external semver validator such as py package [semver](https://pypi.org/project/semver/) to avoid having to test the version regex we've implemented, especially as it becomes increasingly more complex.